### PR TITLE
media-gfx/rawtherapee: version bump to 5.8

### PR DESCRIPTION
**Pull request by vivo75**
Signed-off-by: Francesco Riosa <vivo75@gmail.com>
Closes: https://bugs.gentoo.org/708202

### CODE REVIEW

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+IUSE="openmp tcmalloc"
+
+RDEPEND="
+	dev-libs/expat
+	dev-libs/libsigc++:2
+	media-libs/lcms:2
+	media-libs/lensfun
+	media-libs/libcanberra[gtk3]
+	media-libs/libiptcdata
+	media-libs/libpng:0
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: missing slot op

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	media-libs/tiff:0
+	sci-libs/fftw:3.0
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: missing slot op

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+inherit cmake flag-o-matic toolchain-funcs xdg-utils
+
+DESCRIPTION="A powerful cross-platform raw image processing program"
+HOMEPAGE="https://www.rawtherapee.com/"
+SRC_URI="https://rawtherapee.com/shared/source/${MY_P}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: unnecessary empty line

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	dev-libs/libsigc++:2
+	media-libs/lcms:2
+	media-libs/lensfun
+	media-libs/libcanberra[gtk3]
+	media-libs/libiptcdata
+	media-libs/libpng:0
+	media-libs/tiff:0
+	sci-libs/fftw:3.0
+	sys-libs/zlib
+	tcmalloc? ( dev-util/google-perftools )
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: optional below unconditional deps please

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+IUSE="openmp tcmalloc"
+
+RDEPEND="
+	dev-libs/expat
+	dev-libs/libsigc++:2
+	media-libs/lcms:2
+	media-libs/lensfun
+	media-libs/libcanberra[gtk3]
+	media-libs/libiptcdata
```
Comment (a17r) [inline media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: media-libs/libiptcdata is a py2-only package. Is there a way to make it optional?

Comment (vivo75) [reply media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: I'd say it's difficult to make it optional (deeply embedded in rtexif.{cc,h}) and also still relevant for an imaging software to speak IPTC (used by Adobe extensively)
@heckflosse, @Beep6581 do you have anything to say?

Comment (Beep6581) [reply media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: It is not optional. Ideally we'd like to drop it entirely as libiptcdata is dead, but for the time being it's required.

**Comment by a17r** [comment]
ugh, something went wrong here. please never merge, use rebase instead.

**Comment by vivo75** [comment]
@a17r yes "something" went very wrong, it wasn't a merge but having tried to keep multiple gentoo remotes or something, sorry fort the fuss, should be fixed now

**Comment by vivo75** [comment]
@17r I will create another pull request (after a bit of testing) for media-libs/libiptcdata-1.0.5 - https://bugs.gentoo.org/708706
That one should be accepted _before_ this one

**Comment by vivo75** [comment]
> only waiting for PR14626 then. ;)

Thank you for reminding me, I've totally lost email from your last comment on that PR

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild (hunk 4 above)
Comment (a17r) [reply media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: still not sorted, will fix after merge

**Comment by vivo75** [comment]
> actually none of the review was addressed yet?

changes have probably been lost when I've reverted the git disaster done while trying to cope with both "rsync friendly" gentoo git repo and the actual upstream gentoo repo.
I've redone them and rebased

File: media-gfx/rawtherapee/rawtherapee-5.8.ebuild [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="openmp tcmalloc"
+
+RDEPEND="
+	dev-libs/expat
+	dev-libs/libsigc++:2
+	media-libs/lcms:2
+	media-libs/lensfun:0=
```
[the remark this answers is not in the document]
Comment (vivo75) [reply media-gfx/rawtherapee/rawtherapee-5.8.ebuild]: > I don't see a subslot for lensfun.

then media-gfx/darktable should change too, I've grepped it for the lensfun DEP ;)
give me a few minutes for a new forced push

**Pull request closed without merging**